File: java/src/Position.java

public class Position
{
	private final int MIN_X_POSITION;
	private final int MAX_X_POSITION;
	private final int MIN_Y_POSITION;
	private final int MAX_Y_POSITION;
	
	private int x_position = 0;
	private int y_position = 0;
	
	public Position(int min_x, int max_x, int min_y, int max_y)
	{
		MIN_X_POSITION = min_x;
		MAX_X_POSITION = max_x;
		MIN_Y_POSITION = min_y;
		MAX_Y_POSITION = max_y;
	}
	
	public int getXPosition()
	{
		return x_position;
	}
	
	public boolean setXPosition(int x_position)
	{
		if(x_position >= MIN_X_POSITION && x_position <= MAX_X_POSITION)
		{
			this.x_position = x_position;
			return true;
		}
		return false;
	}
	
	public int getYPosition()
	{
		return y_position;
	}
	
	public boolean setYPosition(int y_position)
	{
		if(y_position >= MIN_Y_POSITION && y_position <= MAX_Y_POSITION)
		{
			this.y_position = y_position;
			return true;
		}
		return false;
	}
	
	public boolean setPosition(int x, int y)
	{
		return ( setXPosition(x) && setYPosition(y) );
	}
}

File: ruby/src/Position.rb
class Position
  @MIN_X_POSITION
  @MAX_X_POSITION
  @MIN_Y_POSITION
  @MAX_Y_POSITION
  
  @x_position = 0
  @y_position = 0
  
  def initialize(min_x, max_x, min_y, max_y)
    @MIN_X_POSITION = min_x
    @MAX_X_POSITION = max_x
    @MIN_Y_POSITION = min_y
    @MAX_Y_POSITION = max_y
  end
  
  def x_position
    return @x_position
  end
  
  def x_position=(x)
    if( x >= @MIN_X_POSITION && x <= @MAX_X_POSITION)
      @x_position = x
      return true
    end
    return false
  end
  
  def y_position
    return @y_position
  end
  
  def y_position=(y)
    if( y >= @MIN_Y_POSITION && y <= @MAX_Y_POSITION)
      @y_position = y
      return true
    end
    return false
  end
  
  def setPosition(x, y)
    self.x_position = x
    self.y_position = y
    return @x_position && @y_position
  end
end
